### PR TITLE
Handle pagination on SearchTweets.scala

### PR DESCRIPTION
**Pull request by jrubin1**
Not perfect code, could be written more functionally, scala-like, stylistically better, etc. but completely working and handles pagination. Handles the API pagination within SearchTweets.scala. I chose that class because I was not sure which endpoint we will end up working with (if we are doing sentiment analysis then looks like we might want an endpoint that gives us tweets that @ a business's twitter account). The pagination functionality can easily be transferred to whatever endpoint we do end up using. This code strictly handles the API pagination, allowing us to access all parts of an API response. The returned JSON strings are currently printed to console. What is not handled in here is: storing request results (because there are multiple ways to store and read the data into Spark that we can choose from). Also important to note that I had to increase the Thread.sleep time to allow time for all requests to come through, and we may want to change the sleep duration based on how many tweets we are receiving from an endpoint.

### CODE REVIEW

File: src/main/scala/SearchTweets.scala [hunk 1]
```diff
@@ -16,24 +16,49 @@ import java.io.IOException;
...
     } else {
       println("There was a problem getting you bearer token. Please make sure you set the BEARER_TOKEN environment variable")
     }
-    Thread.sleep(3000)
+    Thread.sleep(10000)
+  }
+
+  //recursively retrieve all pages from API endpoint, printing them to console
+  def retrievePages(next_token: String): Unit = {
+      var bearerToken: String = "AAAAAAAAAAAAAAAAAAAAAKUnLwEAAAAApWuEw9K8SVH7%2BqNNxVtg36Ryr5o%3DP3sGYZngNVtApjFYqjMC5r3lTg8cF4gGIdCtcjUJAjnd23VpEk"
```
Comment (Trenton-Serpas) [inline src/main/scala/SearchTweets.scala]: A variable bearerToken is already used at line 26, you should pass that variable as a parameter in this function

File: src/main/scala/SearchTweets.scala [hunk 2]
```diff
@@ -16,24 +16,49 @@ import java.io.IOException;
...
+            if (meta.contains("next_token")) {
+              val next_token = meta("next_token")
+              retrievePages(next_token)
+            }
           }
     } else {
       println("There was a problem getting you bearer token. Please make sure you set the BEARER_TOKEN environment variable")
     }
-    Thread.sleep(3000)
+    Thread.sleep(10000)
```
Comment (Trenton-Serpas) [inline src/main/scala/SearchTweets.scala]: Is it possible to change this to an 
```scala
    Await.ready(YOUR_FUTURE_HERE, 10000)
```

**Review by Trenton-Serpas — COMMENTED** [review]
minor tweeks

File: src/main/scala/SearchTweets.scala (hunk 1 above)
Comment (jrubin1) [reply src/main/scala/SearchTweets.scala]: Ah yeah I meant to do this but forgot thanks

File: src/main/scala/SearchTweets.scala (hunk 2 above)
Comment (jrubin1) [reply src/main/scala/SearchTweets.scala]: Yes I could. The code as is is copied from the twitter sample code where they use sleeps. Is it better to use Await.ready? Was also wondering if there is a way to do it/a method where you don't have to specify a sleeping time but not sure if that exists.

Comment (Trenton-Serpas) [reply src/main/scala/SearchTweets.scala]: Await.ready makes it sleep until either the future is finished or the given time is reached so its the best way of achieving that result

Comment (jrubin1) [reply src/main/scala/SearchTweets.scala]: That sounds good. I was also looking at .onComplete() for handling futures- do you know if that one requires a sleep()?

**Review by spark131008 — COMMENTED** [review]
The getTweets method on line 31 still passes only two parameters. 

Also, instead of printing out all response, can you return them into a file formate, so that Trenton can convert it into Dataframe?

**Comment by jrubin1** [comment]
> The getTweets method on line 31 still passes only two parameters.
> 
> Also, instead of printing out all response, can you return them into a file formate, so that Trenton can convert it into Dataframe?

@spark131008 "The getTweets method on line 31 still passes only two parameters." What do you mean by that? What is the significance of the number of parameters passed to getTweets? 

And yes I can do that. As per my first comment on the PR, I did not implement that because there are multiple ways to go about converting it into Spark, and I figured we would want to discuss and agree on a method. If that is the way you guys want to do it, I can do that no problem- and that method of doing it makes sense to me.

File: src/main/scala/SearchTweets.scala (hunk 2 above)
Comment (Trenton-Serpas) [reply src/main/scala/SearchTweets.scala]: onComplete is more for passing a function that runs after the future is ran while Await.ready is closer to a sleep

**Pull request merged**